### PR TITLE
don't prompt for signoff with roles that aren't relevant

### PR DESCRIPTION
**Pull request by CREDO23**
### Changes made:
- Retreive required roles:
    ```javascript
            useEffect(() => {
                const roles =
                  requiredSignoffs.data &&
                  requiredSignoffs.data.data.required_signoffs.map(el => {
                    return el.role;
                  });

                setRequiredRoles(roles);
            }, []);
    ```
- Disable roles that are not required
    ```javascript
        const signoffDialogBody = (
            <FormControl component="fieldset">
              <RadioGroup
                aria-label="Role"
                name="role"
                value={signoffRole}
                onChange={handleSignoffRoleChange}>
                {roles.map(r => {
                  return (
                    <FormControlLabel
                      key={r}
                      value={r}
                      label={r}
                      control={<Radio />}
                      disabled={!(requiredRoles && requiredRoles.includes(r))}
                    />
                  );
                })}
              </RadioGroup>
            </FormControl>
        );
    ```
- If the user has only one role in the required role list, `signoff` directly:
    ```javascript
          const handleSignoff = async rule => {
            const userRequiredRoles = requiredRoles.filter(r => roles.includes(r));

            if (userRequiredRoles.length === 1) {
              const { error, result } = await doSignoff(userRequiredRoles[0], rule);

              if (!error) {
                updateSignoffs(result);
              }
            } else {
              setDialogState({
                ...dialogState,
                open: true,
                title: 'Signoff as…',
                confirmText: 'Sign off',
                item: rule,
                mode: 'signoff',
                handleComplete: handleSignoffDialogComplete,
                handleSubmit: handleSignoffDialogSubmit,
              });
            }
  };
    ```

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -163,9 +163,9 @@ services:
       context: .
       dockerfile: Dockerfile.database
     expose:
-      - "3306"
+      - "3307"
```
Comment (gabrielBusta) [inline docker-compose.yml:166]: This doesn't look related to #1145

Could you revert the changes to this file?

You can keep them on your local, but we don't need to commit/land them.

**Comment by gabrielBusta** [comment]
Thanks for your contribution @CREDO23 I will test it out soon.

File: docker-compose.yml (hunk 1 above)
Comment (CREDO23) [reply docker-compose.yml:166]: Sure,  let me correct this!

**Comment by CREDO23** [comment]
@gabrielBusta, have you tested it ?

**Review by gabrielBusta — APPROVED** [review]

**Comment by CREDO23** [comment]
Now:
- If user's roles are `releng`, `relman`, `tb-releng` and required roles are  `1 relman`, `1 releng` then show a popup so that the user can choose a role to sign with. Here, only  `releng` or `relman` are allowed
- If user's role is `relman` and required roles are  `1 relman`, `1 releng` then `signoff` directly upon update. No need to show a popup
- If user's role is `tb-releng` and required roles are  `1 relman`, `1 releng` then disable the `signoff` button.

**Comment by bhearsum** [comment]
@gabrielBusta - it looks like this is ready for another look?

**Comment by gabrielBusta** [comment]
Thanks @CREDO23!

Works for me on the local instance.

**Comment by AMUZY** [comment]
I actually just tested this and I like it . Great work @CREDO23 .
